pkg/gui/context.go

```diff
@@ -6,20 +6,22 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/jesseduffield/generics/maps"
+	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/gocui"
 	"github.com/jesseduffield/lazygit/pkg/gui/context"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 )
 
 func (gui *Gui) popupViewNames() []string {
-	result := []string{}
-	for _, context := range gui.State.Contexts.Flatten() {
-		if context.GetKind() == types.PERSISTENT_POPUP || context.GetKind() == types.TEMPORARY_POPUP {
-			result = append(result, context.GetViewName())
-		}
-	}
-
-	return result
+	return slices.FilterThenMap(gui.State.Contexts.Flatten(),
+		func(c types.Context) bool {
+			return c.GetKind() == types.PERSISTENT_POPUP || c.GetKind() == types.TEMPORARY_POPUP
+		},
+		func(c types.Context) string {
+			return c.GetViewName()
+		},
+	)
```

Comment:
I might remove FilterThenMap and just opt for a manual filter/map or the FilterMap function. The duplication in the callback method signatures is a little grating.